From e6c57c68618319499571996ef5a8762b9799c1b0 Mon Sep 17 00:00:00 2001
From: Eric Ruei <e-ruei1@ti.com>
Date: Tue, 16 Jun 2015 08:34:53 -0400
Subject: [PATCH] DRM test code enhancements:drm_clone.c drm_extended.c
 get_drm_connector() and get_drm_encoder() Remove the
 HDMI/LCD order assumption

Signed-off-by: Eric Ruei <e-ruei1@ti.com>
---
 drm-tests/drm_clone.c    |   24 ++++++++++++++++++------
 drm-tests/drm_extended.c |   24 ++++++++++++++++++------
 2 files changed, 36 insertions(+), 12 deletions(-)

diff --git a/drm-tests/drm_clone.c b/drm-tests/drm_clone.c
index abbdaea..8b1766c 100644
--- a/drm-tests/drm_clone.c
+++ b/drm-tests/drm_clone.c
@@ -119,19 +119,25 @@ static void get_drm_connector(struct device *dev)
 				dev->hdmi_con = con;
 				printf("hdmi connector id = %d\n",
 					dev->hdmi_con->connector_id);
-				break;
+				if (dev->lcd_con)
+					break;
+				else
+					continue;
 			} else {
 				dev->lcd_con = con;
 				printf("lcd connector id = %d\n",
 					dev->lcd_con->connector_id);
-				continue;
+				if (dev->hdmi_con)
+					break;
+				else
+					continue;
 			}
 		}
 		drmModeFreeConnector(con);
 	}
 
 	if (i == dev->res->count_connectors) {
-		error("No active connector found!\n");
+		error("No active lcd or hdmi connector found!\n");
 		exit(0);
 	}
 }
@@ -157,18 +163,24 @@ static void get_drm_encoder(struct device *dev)
 			dev->lcd_enc = enc;
 			printf("lcd encoder id = %d\n",
 				dev->lcd_enc->encoder_id);
-			continue;
+			if(dev->hdmi_enc)
+				break;
+			else
+				continue;
 		} else if (enc->encoder_id == dev->hdmi_con->encoder_id) {
 			dev->hdmi_enc = enc;
 			printf("hdmi encoder id = %d\n",
 				dev->hdmi_enc->encoder_id);
-			break;
+			if(dev->lcd_enc)
+				break;
+			else
+				continue;
 		}
 		drmModeFreeEncoder(enc);
 	}
 
 	if (i == dev->res->count_encoders) {
-		error("No active encoder found!\n");
+		error("No active lcd or hdmi encoder found!\n");
 		exit(0);
 	}
 }
diff --git a/drm-tests/drm_extended.c b/drm-tests/drm_extended.c
index 93100ef..527ee4a 100644
--- a/drm-tests/drm_extended.c
+++ b/drm-tests/drm_extended.c
@@ -121,19 +121,25 @@ static void get_drm_connector(struct device *dev)
 				dev->hdmi_con = con;
 				printf("hdmi connector id = %d\n",
 					dev->hdmi_con->connector_id);
-				break;
+				if(dev->lcd_con)
+					break;
+				else
+					continue;
 			} else {
 				dev->lcd_con = con;
 				printf("lcd connector id = %d\n",
 					dev->lcd_con->connector_id);
-				continue;
+				if(dev->hdmi_con)
+					break;
+				else
+					continue;
 			}
 		}
 		drmModeFreeConnector(con);
 	}
 
 	if (i == dev->res->count_connectors) {
-		error("No active connector found!\n");
+		error("No active lcd or hdmi connector found!\n");
 		exit(0);
 	}
 }
@@ -159,18 +165,24 @@ static void get_drm_encoder(struct device *dev)
 			dev->lcd_enc = enc;
 			printf("lcd encoder id = %d\n",
 				dev->lcd_enc->encoder_id);
-			continue;
+			if(dev->hdmi_enc)
+				break;
+			else
+				continue;
 		} else if (enc->encoder_id == dev->hdmi_con->encoder_id) {
 			dev->hdmi_enc = enc;
 			printf("hdmi encoder id = %d\n",
 				dev->hdmi_enc->encoder_id);
-			break;
+			if(dev->lcd_enc)
+				break;
+			else
+				continue;
 		}
 		drmModeFreeEncoder(enc);
 	}
 
 	if (i == dev->res->count_encoders) {
-		error("No active encoder found!\n");
+		error("No active lcd or hdmi encoder found!\n");
 		exit(0);
 	}
 }
-- 
1.7.9.5

